skimage/measure/fit.py

```diff
@@ -428,6 +428,20 @@ def estimate(self, data):
         float_type = np.promote_types(data.dtype, np.float32)
         data = data.astype(float_type, copy=False)
 
+        # normalize value range to avoid misfitting due to numeric errors if
+        # the relative distanceses are small compared to absolute distances
+        origin = data.mean(axis=0)
+        data = data - origin
+        scale = data.std()
+        # take care of too low scale values
+        fmin = np.nextafter(0, 1, dtype=float_type)
+        eps = np.spacing(1, dtype=float_type)
+        delta = fmin / eps
+        if scale < delta:
+            # data is a cluster not an ellipse
+            return False
```

Comment:
I am looking for the smallest float where the calculations are still useful/meaningful. The idea here is to have the smallest possible float where I can subtract a number of similar size without loosing precision, i.e. I want to be able to shift the exponent by the size of the precision. For instance, `1.02 - 1.01 = 1e-2`, but if `1e-1` is already the smallest, this would be `0.0`. Therefore, I divide the smallest non-zero float by the precision to add the number of significant digits to the exponent of the smallest non zero float.
I hope the explanation  makes sense to you. If you know any alternative, I would be fine to adopt it.

Just noted, that according to the definition `nextafter(0, 1) == spacing(0)`.

Reply:
Just noted, that this is a standard way of doing it. However, instead of manually calculating it, I will use `np.finfo(float_type).tiny`